packages/vm/src/eei/vmState.ts

```diff
@@ -267,6 +267,7 @@ export class VmState implements EVMStateAccess {
     }
     await this._stateManager.flush()
     // If any empty accounts are put, these should not be marked as touched (when first tx is ran, this account is deleted when it cleans up the accounts)
+    // TODO: not 100% sure if to keep this line or not along cache refactor rebase, 2023-04-11
```

Comment:
I added this in #2621 :) 

We definitely need this. It is an edge case when dumping genesis:

1. Create a genesis with allocation. In this, there is at least one account which is empty. 
2. Now run a block with txs. These txs do not touch the empty account.
3. After the first tx, the account which is empty is cleaned up.

Note that the empty state trie vs. a state trie with 1 empty account has a different state root.

(Removing this line will break `pyspec` hive tests)

Reply:
Ok, have removed my comment.